src/model.js

```diff
@@ -771,6 +771,52 @@ ${associationOwner._getAssociationDebugList()}`);
     });
   }
 
+  /**
+   * Indexes created from options.indexes when calling Model.init
+   */
+  static _manualIndexes;
+
+  /**
+   * Indexes created from {@link ModelAttributeColumnOptions.unique}
+   */
+  static _attributeIndexes;
+
+  static getIndexes() {
+    return [
+      ...(this._manualIndexes ?? []),
+      ...(this._attributeIndexes ?? []),
+      ...(this.uniqueKeys ? Object.values(this.uniqueKeys) : []),
+    ];
+  }
+
+  static get _indexes() {
+    throw new Error('Model._indexes has been replaced with Model.getIndexes()');
+  }
+
+  static _nameIndex(newIndex) {
+    if (Object.prototype.hasOwnProperty.call(newIndex, 'name')) {
+      return newIndex;
+    }
+
+    const newName = Utils.generateIndexName(this.getTableName(), newIndex);
+
+    // TODO: check for collisions on *all* models, not just this one, as index names are global.
```

Comment:
This is for a future PR, right?

Reply:
Yes, it's a lot more complicated than just checking on the same model because they can be scoped by schema, with different rules depending on the database